tests/test_cases/test_cocotb/test_generator_coroutines.py

```diff
@@ -96,14 +96,14 @@ def syntax_error():
     fail  # noqa
 
 
-@cocotb.test(expect_error=True)
+@cocotb.test(expect_error=NameError)
 def test_coroutine_syntax_error(dut):
     """Syntax error in a coroutine that we yield"""
     yield clock_gen(dut.clk)
```

Comment:
This one can be tested with `assert_raises`. Might also want to replace `syntax_error` with `name_error`.

Reply:
I fixed this instance, unfortunately other instances that could otherwise be fixed are abuse the test end behavior of `cocotb.fork` and can't be caught; or aren't in `test_cocotb` and don't have access to `assert_raises`.